account_payment_order/models/account_payment_order.py

```diff
@@ -196,6 +196,23 @@ def check_date_scheduled(self):
                         )
                     )
 
+    @api.constrains("payment_line_ids")
+    def _check_payment_lines(self):
+        for order in self:
+            found_lines = self.env["account.move.line"]
+            for pay_line in order.payment_line_ids:
```

Comment:
This is anti-performance, so it should be avoided. It also adds on each iteration the record to a recordset, having to recreate it and consuming memory. I think a better formula is:

```python
move_line_ids = [x.move_line_id.id for x in order.payment_line_ids]
if len(move_line_ids) != len(set(move_line_ids)):
    raise ...
```